### PR TITLE
Content type updates based on feedback

### PR DESCRIPTION
**Pull request by bseeger**
This updates the content types and taxonomies based on feedback from Katie and Michelle. 

Highlights:
- Lots of minor changes to help text and field ordering. 
- Name changes for Person and Corporate Body name fields:
    - For person: 
       - Person Preferred Name: Primary Part of Name (Required, 1 only), 255 characters
        - Person Preferred Name: 0+ Rest of Name, 255 characters
        - Person Preferred Name: 0+ Prefix, 255 characters
        - Person Preferred Name: 0+ Suffix, 255 characters
        - Person Preferred Name: 0+ Number, 255 characters
        - Person Preferred Name: 0+ Fuller Name, 255 characters 
        - And make a single, repeatable Alternate field:
          - Person Alternate Name (not required, 0+), 1024 characters 
    - For corporate body fields: 
      - Corporate Body: Corporate Name (Required, 1 only) Primary Corporate Name, 255 characters
      - Corporate Body: 0+ Subordinate Name, 255 characters
      - Corporate Body: 0+ Location of Meeting, 255 characters
      - Corporate Body: 0+ Date of meeting or treating signing, 255 characters
      - Corporate Body: 0+ Number of section or meeting, 255 characters
      - And make a single, repeatable alternate field: 
        - Alternate Corporate Body (not required;0+), 1024 characters 
- Tweaks to EDTF formatting - words it uses when displaying intervals and  (in controlled_access_terms)
   -  changes to one error message (years with less then 4 characters need to be padded with zeros)
   -  changes wording of time interval message, when one year is unknown: `1984/..`  will now render as `1984 to unknown`
- Tweaks to where the help text is put (above or below the fields) (in islandora_defaults)
- Updates test code to handle the changes. 
- Hides Help Text field from Authority Links and Url fields 
- On edit forms - changes wording on "Add another item" button to "Add another _field_"

Addresses: https://github.com/jhu-idc/idc-isle-dc/issues/58, https://github.com/jhu-idc/idc-isle-dc/issues/81, https://github.com/jhu-idc/idc-isle-dc/issues/40

### CODE REVIEW

**Comment by bseeger** [comment]
I need to rebase this - as the snapshot is not updated with the latest on dev.

File: codebase/config/sync/migrate_plus.migration.idc_ingest_taxonomy_corporatebody.yml [hunk 1]
```diff
@@ -16,16 +16,47 @@ source:
...
+      plugin: explode
+      delimiter: '|'
   field_primary_name: primary_name
-  field_subordinate_name: subordinate_name
-  field_alt_date_of_meeting: alt_date_of_meeting
-  field_alt_location_of_meeting: alt_location_of_meeting
-  field_alt_num_of_section_or_meet: alt_number_of_section_or_meeting
-  field_alt_primary_name: alt_primary_name
-  field_alt_subordinate_name: alt_subordinate_name
+  field_subordinate_name:
```
Comment (emetsger) [inline codebase/config/sync/migrate_plus.migration.idc_ingest_taxonomy_corporatebody.yml:44]: It looks like `field_subordinate_name` is multi-valued; can an additional value be added to the test CSV to insure this migration pathway is tested?

File: codebase/config/sync/migrate_plus.migration.idc_ingest_taxonomy_persons.yml [hunk 2]
```diff
@@ -20,34 +20,64 @@ source:
...
+      delimiter: '|'
+  field_preferred_name_prefix:
+    -
+      plugin: skip_on_empty
+      method: process
+      source: preferred_name_prefix
+    -
+      plugin: explode
+      delimiter: '|'
+  field_preferred_name_fuller_form:
```
Comment (emetsger) [inline codebase/config/sync/migrate_plus.migration.idc_ingest_taxonomy_persons.yml:40]: It looks like a number of fields were modified to accept multiple values, for example:
`field_preferred_name_rest`, `preferred_name_prefix`, `preferred_name_fuller_form`, etc.

Can the test CSV be updated to include multiple values to ensure these new migration paths are tested?

**Review by emetsger — CHANGES_REQUESTED** [review]
This looks good, tests pass!

My only comment is that there are a number of fields that were modified to accept multiple values, but the test data for these fields contain single values.  If the test data could be updated to contain multiple values, it would insure that these migration pathways are fully tested.


The verification tests would need to be updated as well, to assert the expected number of values equals the actual number, and that the values are equal as well.

**Review by emetsger — APPROVED** [review]
Looks good to me!

**Pull request merged**